platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java

```diff
@@ -314,4 +247,103 @@ private static boolean solrCoreExists(SolrClient client, String coreName)
   }
+
+  private static String[] commaSeparatedToArray(@Nullable String commaDelimitedString) {
+    return (commaDelimitedString != null) ? commaDelimitedString.split("\\s*,\\s*") : new String[0];
+  }
+
+  @Override
+  public org.codice.solr.client.solrj.SolrClient newClient(String core) {
+    String solrUrl =
+        StringUtils.defaultIfBlank(
+            AccessController.doPrivileged(
+                (PrivilegedAction<String>) () -> System.getProperty(SOLR_HTTP_URL)),
+            getDefaultHttpsAddress());
+    final String coreUrl = solrUrl + "/" + core;
+    final String solrDataDir =
+        AccessController.doPrivileged(
+            (PrivilegedAction<String>) () -> System.getProperty(SOLR_DATA_DIR));
+
+    if (solrDataDir != null) {
+      ConfigurationStore.getInstance().setDataDirectoryPath(solrDataDir);
+    }
+    LOGGER.debug("Solr({}): Creating an HTTP Solr client using url [{}]", core, coreUrl);
+    return new SolrClientAdapter(core, () -> createSolrHttpClient(solrUrl, core, coreUrl));
+  }
+
+  @VisibleForTesting
+  SolrClient createSolrHttpClient(String url, String coreName, String coreUrl)
+      throws IOException, SolrServerException {
+    final HttpClientBuilder builder = createHttpBuilder();
+
+    createSolrCore(url, coreName, null, builder.build());
+    try (final Closer closer = new Closer()) {
+      final HttpSolrClient noRetryClient =
+          closer.with(new HttpSolrClient.Builder(coreUrl).withHttpClient(builder.build()).build());
+      final HttpSolrClient retryClient =
+          closer.with(
+              new HttpSolrClient.Builder(coreUrl)
+                  .withHttpClient(
+                      builder.setRetryHandler(new SolrHttpRequestRetryHandler(coreName)).build())
+                  .build());
+
+      return closer.returning(new PingAwareSolrClientProxy(retryClient, noRetryClient));
+    }
+  }
+
+  public boolean useTls() {
+    return StringUtils.startsWithIgnoreCase(
+        AccessController.doPrivileged(
+            (PrivilegedAction<String>) () -> System.getProperty(SOLR_HTTP_URL)),
+        "https");
+  }
+
+  private HttpClientBuilder createHttpBuilder() {
+
+    HttpClientBuilder httpClientBuilder =
+        HttpClients.custom()
+            .setDefaultCookieStore(new BasicCookieStore())
+            .setMaxConnTotal(128)
+            .setMaxConnPerRoute(32);
+
+    if (useTls()) {
+      httpClientBuilder.setSSLSocketFactory(
+          new SSLConnectionSocketFactory(
+              getSslContext(),
+              DEFAULT_PROTOCOLS,
+              DEFAULT_CIPHER_SUITES,
+              SSLConnectionSocketFactory.BROWSER_COMPATIBLE_HOSTNAME_VERIFIER));
+    }
+
+    boolean useBasicAuth =
+        AccessController.doPrivileged(
+            (PrivilegedAction<Boolean>)
+                () -> Boolean.valueOf(System.getProperty("solr.useBasicAuth")));
+    if (useBasicAuth) {
+      // TODO: When this JAR becomes a bundle, convert SolrPasswordUpdate to become a bean.
```

Comment:
Should we just create a ticket to do this and remove the TODO?

Reply:
In retrospect, I don't think the comment is necessary. I don't think there is a ticket for converting solr-factory into a bundle. It was a conversation I had with @tbatie. I think the comment can be removed.

Reply:
There is one and it is being worked on by @tbatie right now.